src/lib/data/alignment-group.js

```diff
@@ -34,10 +47,27 @@ export default class AlignmentGroup {
   }
 
   isFirstToken (token) {
-    return this.steps.length > 0 && this.steps[0].id === token.idWord
+    return this.firstStep.id === token.idWord
   }
 
   tokenTheSameTextTypeAsStart (token) {
-    return this.steps.length > 0 && this.steps[0].textType === token.textType
+    return this.steps.length > 0 && this.firstStep.textType === token.textType
+  }
+
+  updateFirstStep (token) {
+    this.firstStep = { textType: token.textType, id: token.idWord }
+  }
+
+  merge (tokensGroup) {
+    this.origin.push(...tokensGroup.origin)
+    this.target.push(...tokensGroup.target)
+
+    tokensGroup.origin.forEach(idWord => {
+      this.steps.push({ textType: 'origin', id: idWord, type: 'merge' })
+    })
+
+    tokensGroup.target.forEach(idWord => {
+      this.steps.push({ textType: 'target', id: idWord, type: 'merge' })
+    })

```

Comment:
I agree that the way you manage the steps will have to be updated to deal with undo of a merge -- i.e you'll need a way to indicate which actions should be undone/redone together.

Reply:
yes, I need to try some variants